src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XmlUtil.java

```diff
@@ -132,4 +132,30 @@ else if (child.hasChildNodes()) {
         return result;
     }
 
+    /**
+     * Returns the value of the "name" attribute for the given node.
+     *
+     * @param node to retrieve the name
+     * @return the node name
+     */
+    public static String getNodeName(Node node) {
+        return node.getAttributes().getNamedItem("name").getNodeValue();
+    }
+
+    /**
+     * Removes all whitespaces at the beginning and at the end of the string and
+     * replaces repeated whitespaces in the middle of the string with a single space.
+     * Replaces XML entities with escaped values.
```

Comment:
The first sentence should be the summary, but you have it split into 2 and is a bit long.

I think the first sentence should be something like "This santizes the given string for safe use in XML documents." and then list what it specifically does as consecutive sentences.

Reply:
done